models/classes/user/UserSettingsServiceProvider.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA.
+ */
+
+declare(strict_types=1);
+
+namespace oat\tao\model\user;
+
+use oat\generis\model\data\Ontology;
+use oat\generis\model\DependencyInjection\ContainerServiceProviderInterface;
+use oat\oatbox\user\UserTimezoneServiceInterface;
+use oat\tao\model\user\implementation\UserSettingsService;
+use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+
+use function Symfony\Component\DependencyInjection\Loader\Configurator\service;
+
+class UserSettingsServiceProvider implements ContainerServiceProviderInterface
+{
+    public function __invoke(ContainerConfigurator $configurator): void
+    {
+        $services = $configurator->services();
+
+        $services
+            ->set(UserSettingsServiceInterface::class, UserSettingsService::class)
```

Comment:
I recommend not use the interface as service Identifier, cause when we have multiple implementations the interface will be already hardcoded to this one. The interface should be just type hinted in the client class to follow open/closed. Wdyt?

```suggestion
            ->set(UserSettingsService::class, UserSettingsService::class)
```

Reply:
Then we'll need to bind `tao_actions_UserSettings` to `UserSettingsService` instead of `UserSettingsServiceInterface`.

That is, we'll be getting the service by asking the PSR container for the particular implementation instead of the interface: `$this->getPsrContainer()->get(UserSettingsService::class);` instead of `$this->getPsrContainer()->get(UserSettingsServiceInterface::class);`.

That's ok?

Reply:
It is ok :). It is your call, not a problem for me any solution in this case.

Reply:
Changed: 616291e818a1d222696f0fd1940a7dcd3cd53772